### DISCUSSION TITLE
Improve the documentation for QueryClient options

### DISCUSSION THREAD

**Discussion by ghost** [TanStack/query#2412] [General]
Whilst trying out this library I've found the documentation quite nice to work with, with one exception: the way the options for the QueryClient are documented (see: https://react-query.tanstack.com/reference/QueryClient).

If you look under options you'll see:

<img width="598" alt="Screenshot 2021-06-29 at 12 04 41" src="https://user-images.githubusercontent.com/7355199/123779194-35117380-d8d2-11eb-9d51-32e6c33064e8.png">

However, the options themselves aren't linked from that section, and I couldn't find them in the docs anywhere. In the end I ended up going through the source and sort of cobbling it together myself. In my opinion this could be improved by either listing the available options, or linking to a list like that in the docs.

**Comment by TkDodo** [comment]
- `QueryCache` should link here: https://react-query.tanstack.com/reference/QueryCache
- `MutationCache` should link here: https://react-query.tanstack.com/reference/MutationCache

It would be cool if you could PR that

**Reply by ghost** [reply]
@TkDodo Thanks for the quick reply. And where should the `defaultOptions` link? Because that was the main thing that took quite some time to find (and I didn't end up finding a good source for it in the end).

**Reply by TkDodo** [reply]
I think we'd need to add them, maybe directly to this page?

The TypeScript types will probably be most helpful:

https://github.com/tannerlinsley/react-query/blob/3b2c4088e103e4b4ca990662c03fd6e67330b477/src/core/types.ts#L563-L566

Generally, for queries, you can pass in anything that `useQuery` accepts as `options`, and for mutations it's whatever `useMutation` accepts. Maybe that's enough of an explanation, and we could link to the respective API docs then?

**Reply by ghost** [reply]
> Generally, for queries, you can pass in anything that useQuery accepts as options, and for mutations it's whatever useMutation accepts. Maybe that's enough of an explanation, and we could link to the respective API docs then?

Yeah I think that that's a good solution 👍. I'll see if I have time somewhere and submit a PR if I do. Otherwise, if someone else feels like tackling this feel free to do so!
